src/pages/RegistrationForm/RegistrationForm.tsx

```diff
@@ -38,69 +44,137 @@ const NewPage: React.FC = () => {
         }),
-        onSubmit: values => {
-            console.log('Form data', values);
-            formik.resetForm();
-            setSelectedDate(null);
-            setIsRegistrationSuccessful(true);
-        }
 
+        onSubmit: (values: FormValues) => {
+            if (hasEmptyField(values)) {
+                toast.error('One or more fields are empty. Please fill in all required fields.');
+            } else {
+                console.log('Form data', values);
+                formik.resetForm();
+                toast.success('Registration successfully completed!');
+            }
+        }
     });
+    const hasEmptyField = (values: FormValues) => {
+        const allValues = Object.values(values);
+        return allValues.some(value => !value);
+    };
+
     return (
-        <div className="bg-gray-100 min-h-screen">
+        <div className="min-h-screen">
             <Header/>
             <div className="my-6 border-t border-gray-200"></div>
-            <h2 className="text-2xl mb-4 font-semibold text-green-500">Sign up</h2>
-            {isRegistrationSuccessful && (
-                <div className="text-green-500 mt-2">Регистрация успешно завершена!</div>
-            )}
+            <h2 className="text-2xl mb-4 font-semibold text-blue-700">Sign up</h2>
+            <ToastContainer/>
             <form onSubmit={formik.handleSubmit} className="mt-6 space-y-4 flex justify-center flex-col m-auto w-max">
-                <div className="flex flex-col w-72">
+                <div className="relative h-10 w-full min-w-[200px] mt-5">
                     <input
-                        placeholder="First name"
-                        className="border rounded py-2 px-3 focus:border-indigo-500"
+                        className=" peer h-full w-full rounded-[7px] border border-blue-gray-200 border-t-transparent
+                         bg-transparent px-3 py-2.5 font-sans text-sm font-normal text-blue-gray-700 outline outline-0
+                          transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200
+                           placeholder-shown:border-t-blue-gray-200 focus:border-2 focus:border-indigo-500
+                            focus:border-t-transparent focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50"
+                        placeholder=" "
                         id="firstName"
                         name="firstName"
                         type="text"
                         onChange={formik.handleChange}
                         onBlur={formik.handleBlur}
                         value={formik.values.firstName}
                     />
-                    {formik.touched.firstName && formik.errors.firstName ? (
-                        <div className="text-red-500 mt-1">{formik.errors.firstName}</div>
-                    ) : null}
+                    <label className="before:content[' '] after:content[' '] pointer-events-none absolute left-0 -top-1.5
```

Comment:
wow why you use a lot of classes

Reply:
One of Tailwind's idea of starting with utilities and extracting common patterns as you notice them. Consider extracting commonly used combinations of utility classes into custom component classes.

something like @apply

.label-base {
  @apply text-blue-gray-400 transition-all;
  /* other styles */
}

and now you can use this class in diffrent places

you can add it to css file